lib/ash_admin/router.ex

```diff
@@ -90,6 +90,8 @@ defmodule AshAdmin.Router do
   def __session__(conn, [session]), do: __session__(conn, session)
 
   def __session__(conn, session) do
+    session = Map.put(session, "prefix", conn.request_path)
```

Comment:
So this is where we need to add the code, but unfortunately, I'm pretty sure `conn.request_path` is not going to work. If you were to go to `/admin/foo/bar` then `conn.request_path` would be `/admin/foo/bar` IIRC.

Reply:
That is right, yes. But that was the problem, isn't it? The whole path wasn't visible

Reply:
The "prefix" in that example needs to be `"/admin"`. We use this to know how to build links internally. So if we want to link to `/foo` from inside the admin dashboard, we need to link to `/admin/foo`. So we add the prefix to any links we generate.

If they did
```elixir
scope "/foo" do
  scope "/bar" do
     admin "/admin"
  end
end
```

then the prefix would need to be `"/foo/bar/admin"`.

Reply:
In that last example I gave, if they went to `/foo/bar/admin/baz` we'd need `prefix` to be `/foo/bar/admin` still.

Reply:
I think I got you 

How does this look in the mount?

```elixir
    prefix =
      case prefix do
        "/" ->
          prefix

        _ ->
          request_path = session["request_path"]
          string_length = String.length(request_path) - String.length(prefix)
          <<path_prefix::binary-size(string_length), _rest::binary>> = request_path
          path_prefix <> prefix
      end

```

Reply:
Ah, no. Got your new comment after refreshing the page :(

Reply:
Okay, this

```elixir
    prefix =
      case prefix do
        "/" ->
          prefix

        _ ->
          request_path = session["request_path"]
          [scope, _] = String.split(path, prefix)
          scope <> prefix
      end
```

Reply:
The `prefix` in `case prefix do` is the one from the pattern match on the function head of `mount`

Reply:
🤔 I'm pretty sure this still requires the user explicitly passing a `prefix` option. The `on_mount` hook uses the `"prefix"` value from the session, which is set by the user doing `path: "/something"`, right?